internal/api/middleware/timer.go

```diff
@@ -0,0 +1,46 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi"
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/dogstatsd"
+)
+
+func getHist(r *http.Request, t *dogstatsd.Timing) metrics.Histogram {
+	str := strings.ReplaceAll(chi.RouteContext(r.Context()).RoutePattern(), "/", ".")
+	fmt.Println(str)
+	regx := regexp.MustCompile(`[{}]`)
+	if regx.MatchString(str) {
+		str = regx.ReplaceAllString(str, "")
+	}
+
+	str = strings.Replace(str, ".", "", 1)
+
+	fmt.Println(str)
+	return t.With(str, "method", r.Method)
```

Comment:
The metric label tags must always be provided in pairs, but only 3 are provided here. This is should look like `t.With("endpoint", str, "method", r.Method)`